File: Examples/VisualBasic/WorkingWithTasks/WriteTaskProperties.vb
Imports Aspose.Tasks.Saving

'
'This project uses Automatic Package Restore feature of NuGet to resolve Aspose.Tasks for .NET API reference 
'when the project is build. Please check https://docs.nuget.org/consume/nuget-faq for more information. 
'If you do not wish to use NuGet, you can manually download Aspose.Tasks for .NET API from http://www.aspose.com/downloads, 
'install it and then add its reference to this project. For any issues, questions or suggestions 
'please feel free to contact us using http://www.aspose.com/community/forums/default.aspx
'

Namespace WorkingWithTasks
    Public Class WriteTaskProperties
        Public Shared Sub Run()
            ' ExStart:WriteTaskProperties
            ' Create project instance
            Dim project As New Project()

            ' Add task and set task properties
            Dim task As Task = project.RootTask.Children.Add("Task1")
            task.[Set](Tsk.Start, project.RootTask.[Get](Tsk.Start).AddDays(1))
            task.[Set](Tsk.Name, "new name")
            ' ExEnd:WriteTaskProperties

            ' Save project
            Dim dataDir As String = RunExamples.GetDataDir(System.Reflection.MethodBase.GetCurrentMethod().DeclaringType)
            project.Save(dataDir + "project_out.pdf", SaveFileFormat.PDF)
        End Sub
    End Class
End Namespace

File: Examples/CSharp/WorkingWithTasks/WriteTaskProperties.cs
/*
This project uses Automatic Package Restore feature of NuGet to resolve Aspose.Tasks for .NET API reference 
when the project is build. Please check https:// Docs.nuget.org/consume/nuget-faq for more information. 
If you do not wish to use NuGet, you can manually download Aspose.Tasks for .NET API from https://www.nuget.org/packages/Aspose.Tasks/, 
install it and then add its reference to this project. For any issues, questions or suggestions 
please feel free to contact us using https://forum.aspose.com/c/tasks
*/

namespace Aspose.Tasks.Examples.CSharp.WorkingWithTasks
{
    using Aspose.Tasks.Saving;

    internal class WriteTaskProperties
    {
        public static void Run()
        {
            var dataDir = RunExamples.GetDataDir(System.Reflection.MethodBase.GetCurrentMethod());
            
            //ExStart:WriteTaskProperties
            //ExFor: Tsk.Start
            //ExFor: Tsk.Name
            //ExSummary: Shows how to set task properties.
            var project = new Project();
            
            // Add task and set task properties
            var task = project.RootTask.Children.Add("Task1");
            task.Set(Tsk.Start, project.RootTask.Get(Tsk.Start).AddDays(1));
            task.Set(Tsk.Name, "Task1");
            
            // Save project
            project.Save(dataDir + "WriteTaskProperties_out.pdf", SaveFileFormat.PDF);
            //ExEnd:WriteTaskProperties
        }
    }
}
